### PR TITLE
Adds tests for distributed closest point query

### LINKED ISSUE #884: Update MacOS version in Azure pipelines
I noticed this warning in our Azure pipelines:

> ##[warning]The macOS-10.15 environment is deprecated, consider switching to macos-11(macos-latest), macos-12 instead. For more details see https://github.com/actions/virtual-environments/issues/5583

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR is a (minor) feature
- It adds unit tests for the distributed closest point query in 2D for each enabled execution policy
- It is in support of #882 
- This PR should be merged before #879 to ensure that the distributed closest point functionality works for each policy and host-config before and after that PR

#### Misc:
- Fixes #884 by updating our azure CI config to `macos-11`
- Changes the file creation mode in the `makeDirsForPath()` file utility function from `0770` to `0777` to resolve a file error (`EACCESS`) in our docker-based CI tests
- Loosens checks for the upper bound in the `random_real` unit tests. Although `std::uniform_real_distribution` returns a value in `[0,1)` it seems that the result might be equal to the upper bound after transforming to the range `[min_value, max_value)`

### CODE REVIEW

File: src/axom/quest/examples/quest_distributed_distance_query_example.cpp [hunk 1]
```diff
@@ -57,6 +57,8 @@ struct Input
 {
 public:
   std::string meshFile;
+  std::string distanceFile {"closest_point"};
```
Comment (kennyweiss) [inline src/axom/quest/examples/quest_distributed_distance_query_example.cpp:60]: I borrowed the changes in this file from #879 and attributed the commit to @gunney1

**Review by white238 — APPROVED** [review]

**Comment by rhornung67** [comment]
> > @kennyweiss one question: should there be tests for meshes where the number of elements along each coordinate direction are not the same; i.e., would this help catch issues that working on logically square/cubic meshes do not?
> 
> Sure. That would be easy enough to do. What resolution(s) were you thinking? The test is currently 30x30 to keep runtimes reasonably low for our unit tests. Do you think 20 x 40 would capture this difference? Or were you thinking of some other resolution(s)?

My thinking was not specific regarding problem size. It just struck me that all the tests are on square/cubic meshes. I defer to your judgment whether non-square/cubic mesh tests would ensure additional robustness.

**Comment by gunney1** [comment]
Thank you @kennyweiss

**Review by gunney1 — APPROVED** [review]

**Comment by kennyweiss** [comment]
Running into some problems with writing files to our docker images and expecting to read them. 
I'm going to put those files in our data repo and use them for the tests instead.

**Pull request merged**